ext/sockets/sockets.c

```diff
@@ -2089,7 +2097,15 @@ PHP_FUNCTION(socket_create_pair)
 		RETURN_THROWS();
 	}
 
-	if (type > 10) {
+	stype = type;
+#ifdef SOCK_CLOEXEC
+	stype &= ~SOCK_CLOEXEC;
+#endif
+#ifdef SOCK_NONBLOCK
+	stype &= ~SOCK_NONBLOCK;
+#endif
+
+	if (stype > 10) {
 		zend_argument_value_error(2, "must be one of SOCK_STREAM, SOCK_DGRAM, SOCK_SEQPACKET,"
```

Comment:
Wouldn't make sense to update this message so it is clear that `SOCK_CLOEXEC` and `SOCK_NONBLOCK` are not supported. I guess they are always either both supported or none so one ifdef should suffic.

Reply:
Not sure I follow, what message do you have in mind ? we re just checking socket type here.

Reply:
```
must be one of SOCK_STREAM, SOCK_DGRAM, SOCK_SEQPACKET, SOCK_RAW, or SOCK_RDM
```
After your change `SOCK_CLOEXEC` and `SOCK_NONBLOCK` is also allowed so was thinking that this message is not exactly correct.